src/runtime/capi.cpp

```diff
@@ -1872,7 +1872,8 @@ Box* BoxedCApiFunction::tppCall(Box* _self, CallRewriteArgs* rewrite_args, ArgPa
         rewrite_args->out_success = true;
     }
 
-    checkAndThrowCAPIException();
+    if (!rtn)
+        checkAndThrowCAPIException();
```

Comment:
tiny, but this should be throwCAPIException() (in case the user returned NULL but didn't set an exception).